app/src/main/java/ca/carleton/gcrc/n2android_mobile1/connection/ConnectionSyncProcess.java

```diff
@@ -432,16 +502,107 @@ public Request createAttachmentRequest(String uploadId, String filepath, String
+            return submissionStatusByDocId.get(docId);
+        } else {
+            return SubmissionStatus.NotSubmitted;
+        }
+    }
+
+    private boolean IsDocumentDeletedOnRemote(String docId) throws Exception {
+        Request request = createDocumentStatusRequest(docId);
+        Response response = submissionClient.newCall(request).execute();
+
+        JSONObject jsonResult = new JSONObject(response.body().string());
+
+        return jsonResult.has("error") && jsonResult.has("reason") && jsonResult.optString("reason").equals("deleted");
+    }
+
+    private JSONObject getSubmissionStatus() throws Exception {
+        Response response = submissionClient.newCall(createSubmissionStatusRequest()).execute();
+        String body = response.body().string();
+        Log.v(TAG, body);
+
+        if (!response.isSuccessful()) {
+            throw new RuntimeException("Creating new database document failed: " + body);
+        }
+
+        return new JSONObject(body);
+    }
+
+    private Request createSubmissionStatusRequest() throws Exception {
+        ConnectionInfo info = connection.getConnectionInfo();
+
+        // Put the item in the submission database.
+        HttpUrl url = HttpUrl.parse(info.getUrl()).newBuilder().encodedPath("/servlet/submission/_submission-info-by-device-id").addQueryParameter("key", info.getDeviceId()).build();
+
+        return new Request.Builder()
+                .url(url)
+                .get()
+                .build();
+    }
+
+    private Request createDocumentStatusRequest(String documentId) throws Exception {
+        ConnectionInfo info = connection.getConnectionInfo();
+
+        // Put the item in the submission database.
+        HttpUrl url = HttpUrl.parse(info.getUrl()).newBuilder().encodedPath("/db/" + documentId).build();
+
+        return new Request.Builder()
+                .url(url)
+                .get()
+                .build();
+    }
+
+    private Map<String, SubmissionStatus> createSubmissionStatusMap(JSONObject submissionStatus) throws Exception {
+        Map<String, SubmissionStatus> submissionStatusMap = new HashMap<>();
+
+        JSONArray rows = submissionStatus.getJSONArray("rows");
+        for (int i=0;i<rows.length();i++) {
+            JSONObject record = rows.getJSONObject(i);
+            String docId = record.getJSONObject("value").getString("docId");
+            String state = record.getJSONObject("value").getString("state");
+
+            SubmissionStatus status = SubmissionStatus.Unknown;
+
+            switch (state) {
+                case "waiting_for_approval":
+                    status = SubmissionStatus.WaitingForApproval;
+                    break;
+                case "approved":
+                case "complete":
+                    status = SubmissionStatus.Completed;
+                    break;
+                case "denied":
+                    status = SubmissionStatus.Declined;
+                    break;
```

Comment:
Should we have a default that throws an exception? Or are we guaranteed a valid state?

Reply:
So we have to do this check on all of the documents, since we currently don't have a record of whats been submitted (this is something we might have to add). So I need to do this check on all of the documents so I can't currently know whats in an invalid state and whats just not submitted yet. Interested in your thoughts.

Reply:
Ok, that's fine. Is it possible for `state` to be null? In this case, this function would already throw an JSON Parsing exception.

Reply:
AFAIK, it is not possible, but I should guard against it. I'll fix that.